src/api_server/src/request/snapshot.rs

```diff
@@ -43,10 +54,65 @@ pub(crate) fn parse_patch_vm_state(body: &Body) -> Result<ParsedRequest, Error>
     }
 }
 
+fn parse_put_snapshot_load(body: &Body) -> Result<ParsedRequest, Error> {
+    let snapshot_config =
+        serde_json::from_slice::<LoadSnapshotConfig>(body.raw()).map_err(Error::SerdeJson)?;
+
+    // Ensure `mem_file_path` and `mem_backend` fields are not present at the same time.
+    if snapshot_config.mem_backend.is_some() && snapshot_config.mem_file_path.is_some() {
+        return Err(Error::SerdeJson(serde_json::Error::custom(TOO_MANY_FIELDS)));
+    }
+    // Ensure that one of `mem_file_path` or `mem_backend` fields is always specified.
+    if snapshot_config.mem_backend.is_none() && snapshot_config.mem_file_path.is_none() {
+        return Err(Error::SerdeJson(serde_json::Error::custom(MISSING_FIELD)));
+    }
+
+    // Check for the presence of deprecated `mem_file_path` field and create
+    // deprecation message if found.
+    let mut deprecation_message = None;
+    if snapshot_config.mem_file_path.is_some() {
+        // `mem_file_path` field in request is deprecated.
+        METRICS.deprecated_api.deprecated_http_api_calls.inc();
+        deprecation_message = Some(LOAD_DEPRECATION_MESSAGE);
+    }
+
+    // If `mem_file_path` is specified instead of `mem_backend`, we construct the
+    // `MemBackendConfig` object from the path specified, with `File` as backend type.
+    let mem_backend = match snapshot_config.mem_backend {
+        Some(backend_cfg) => backend_cfg,
+        None => {
+            MemBackendConfig {
+                // This is safe to unwrap() because we ensure above that one of the two:
+                // either `mem_file_path` or `mem_backend` field is always specified.
+                backend_path: snapshot_config.mem_file_path.unwrap(),
+                backend_type: MemBackendType::File,
+            }
+        }
+    };
+
+    let snapshot_params = LoadSnapshotParams {
+        snapshot_path: snapshot_config.snapshot_path,
+        mem_backend,
+        enable_diff_snapshots: snapshot_config.enable_diff_snapshots,
+        resume_vm: snapshot_config.resume_vm,
+    };
+
+    // Construct the `ParsedRequest` object.
+    let mut parsed_req = ParsedRequest::new_sync(VmmAction::LoadSnapshot(snapshot_params));
+
+    // If `mem_file_path` was present, set the deprecation message in `parsing_info`.
+    if let Some(msg) = deprecation_message {
+        parsed_req.parsing_info().append_deprecation_message(msg);
```

Comment:
I think we could do this with `Option::map`.

Reply:
Good suggestion! Done.